src/provider.rs

```diff
@@ -138,32 +194,77 @@ impl StatusLineElement {
+    pub fn new<I: Iterator<Item = Element>, J: Iterator<Item = Replacement>>(
+        // we get the arguments when we're called
+        statusline_elements_iter: I,
+        line_trans_iter: J,
+        separator: String,
+    ) -> RahmenResult<Self> {
+        // read the metadata config entries and store them to the elements vector
         let mut elements = vec![];
-        for element in elements_iter {
+        for element in statusline_elements_iter {
             elements.push(element.try_into()?);
         }
-        Ok(Self { elements })
+        // read the postprocessing regexes and store them to the line_transformations vector
+        let mut line_transformations = vec![];
+        for line_transform in line_trans_iter {
+            //println!("LT: {:?}", line_transform);
+            line_transformations.push(line_transform.try_into()?);
+        }
+        // return the vector(s)
+        Ok(Self {
+            elements,
+            line_transformations,
+            separator,
+        })
     }
 
-    /// Format the meta data from the given path using this formatter
+    /// Format the meta data from the given path (called as an adaptor to the status line formatter)
     pub fn format<P: AsRef<std::ffi::OsStr>>(&self, path: P) -> RahmenResult<String> {
         let metadata = Metadata::new_from_path(path)?;
-        // iterate over the tag table
-        Ok(self
+        // iterate over the tag vector we built in the constructor, but stop when we have an
+        // iterator of strings
+        //let mut status_line = self
+        let mut element_iter = self
             .elements
             .iter()
+            // process each metadata section (element) using the associated transformation instructions
             .flat_map(move |element| element.process(&metadata))
-            // remove multiples (e.g. if City and  ProvinceState are the same)
-            .unique()
-            .join(", "))
+            // remove empty strings (which may be the result of a transformation regex replacement)
+            .filter(|x| !x.is_empty());
+        // TODO discuss if this should better be made configurable (possibly so)
+        // when there is no postprocessing regex...
+        let mut status_line = if self.line_transformations.iter().peekable().peek().is_none() {
+            // ...remove multiples (e.g. if City and  ProvinceState are the same) and
+            // join the strings using the separator
+            element_iter.unique().join(&self.separator)
+        } else {
+            // ...else only create a joint string ('line')
+            element_iter.join(&self.separator)
+        };
+
+        for transformation in &self.line_transformations {
+            status_line = transformation.transform(&status_line);
+        }
+
+        // you could write the above for... statement like this, too
+        /*
+        status_line = self.line_transformations.iter().fold(status_line.clone(), |sl, t| t.transform(sl));
```

Comment:
Thinking about it, this seems to be the more elegant approach. To avoid copies, use:

```
        let status_line = self.line_transformations.iter().fold(status_line, |sl, t| t.transform(sl));
```
(Added `let`, removed `clone`)